Sources/SwiftDocC/Model/Rendering/Content/RenderBlockContent.swift

```diff
@@ -429,6 +462,102 @@ public enum RenderBlockContent: Equatable {
+            case "metadata": self = .metadata
+            case "colspan": self = .colspan
+            case "rowspan": self = .rowspan
+            default:
+                let coordinates = stringValue.split(separator: "_")
+                guard coordinates.count == 2,
+                      let rowIndex = Int(coordinates.first!),
+                      let columnIndex = Int(coordinates.last!) else {
+                    return nil
+                }
+                self = .index(row: rowIndex, column: columnIndex)
+            }
+        }
+
+        var intValue: Int? { nil }
+
+        init?(intValue: Int) {
+            return nil
+        }
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+
+        var extendedData = Set<RenderBlockContent.TableCellExtendedData>()
+        if container.allKeys.contains(.extendedData) {
+            let dataContainer = try container.nestedContainer(keyedBy: CodingKeys.self, forKey: .extendedData)
+
+            for index in dataContainer.allKeys {
+                guard case let .index(row, column) = index else { continue }
+
+                let cellContainer = try dataContainer.nestedContainer(keyedBy: CodingKeys.self, forKey: index)
+                extendedData.insert(.init(rowIndex: row,
+                                          columnIndex: column,
+                                          colspan: try cellContainer.decode(UInt.self, forKey: .colspan),
+                                          rowspan: try cellContainer.decode(UInt.self, forKey: .rowspan)))
+            }
+        }
+
+        self = .init(header: try container.decode(RenderBlockContent.HeaderType.self, forKey: .header),
+                     rows: try container.decode([RenderBlockContent.TableRow].self, forKey: .rows),
+                     extendedData: extendedData,
+                     metadata: try container.decodeIfPresent(RenderContentMetadata.self, forKey: .metadata))
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        try container.encode(header, forKey: .header)
+        try container.encode(rows, forKey: .rows)
+
+        if !extendedData.isEmpty {
+            var dataContainer = container.nestedContainer(keyedBy: CodingKeys.self, forKey: .extendedData)
+            for data in extendedData {
+                var cellContainer = dataContainer.nestedContainer(keyedBy: CodingKeys.self,
+                                                                  forKey: .index(row: data.rowIndex,
+                                                                                 column: data.columnIndex))
+                try cellContainer.encode(data.colspan, forKey: .colspan)
+                try cellContainer.encode(data.rowspan, forKey: .rowspan)
+            }
+        }
+
+        try container.encodeIfPresent(metadata, forKey: .metadata)
+    }
+}
```

Comment:
It's really only `TableCellExtendedData` that needs a custom `Codable` implementation, the rest of `Table` can use the default synthesized implementation. This shortens the custom implementation and focuses it on the custom behavior. 

```suggestion
extension RenderBlockContent.Table: Codable { }

// Writing a manual Codable implementation for table's extended data because the encoding doesn't match the struct layout.
// Instead the row and column indices are encoded as a dynamic key to another keyed container that hold the row and column spans.
extension RenderBlockContent.TableCellExtendedData: Codable {

    // TableCellExtendedData encodes the row and column indices as a dynamic key with the format "{row}_{column}".
    struct DynamicIndexCodingKey: CodingKey, Equatable {
        let row, column: Int
        init(row: Int, column: Int) {
            self.row = row
            self.column = column
        }
        
        var stringValue: String {
            return "\(row)_\(column)"
        }

        init?(stringValue: String) {
            let coordinates = stringValue.split(separator: "_")
            guard coordinates.count == 2,
                  let rowIndex = Int(coordinates.first!),
                  let columnIndex = Int(coordinates.last!) else {
                return nil
            }
            row = rowIndex
            column = columnIndex
        }

        // The key is only represented by a string value
        var intValue: Int? { nil }
        init?(intValue: Int) { nil }
    }
    
    enum ExtendedDataCodingKeys: String, CodingKey {
        case colspan, rowspan
    }
    
    public init(from decoder: Decoder) throws {
        let container = try decoder.container(keyedBy: DynamicIndexCodingKey.self)
        guard let indexKey = container.allKeys.first else {
            throw DecodingError.keyNotFound(DynamicIndexCodingKey(row: 0, column: 0),
                                            DecodingError.Context(codingPath: container.codingPath, debugDescription: "Missing key with format \"{row}_{column}\"."))
        }
        
        let dataContainer = try container.nestedContainer(keyedBy: ExtendedDataCodingKeys.self, forKey: indexKey)
        
        self.rowIndex = indexKey.row
        self.columnIndex = indexKey.column
        self.rowspan = try dataContainer.decode(UInt.self, forKey: .rowspan)
        self.colspan = try dataContainer.decode(UInt.self, forKey: .colspan)
    }
    
    public func encode(to encoder: Encoder) throws {
        var container = encoder.container(keyedBy: DynamicIndexCodingKey.self)
        var dataContainer = container.nestedContainer(keyedBy: ExtendedDataCodingKeys.self, forKey: .init(row: rowIndex, column: columnIndex))
        try dataContainer.encode(colspan, forKey: .colspan)
        try dataContainer.encode(rowspan, forKey: .rowspan)
    }
}
```

Reply:
This still encodes with the expected format:
```
"extendedData" : [
  {
    "1_0" : {
      "colspan" : 2,
      "rowspan" : 2
    }
  },
  {
    "2_0" : {
      "colspan" : 2,
      "rowspan" : 0
    }
  },
  {
    "1_1" : {
      "colspan" : 0,
      "rowspan" : 1
    }
  },
  {
    "2_1" : {
      "colspan" : 0,
      "rowspan" : 1
    }
  }
]
```

Reply:
I like that a lot! I didn't know that that was possible, since i didn't realize that the `Codable` implementations could access the key directly like that.

It looks like this causes a problem when decoding a table that *doesn't* have `extendedData`, though, so i wound up having to make the `extendedData` set optional in the same breath. That should be fine, but it's something else that i needed to add to make tests pass.

Reply:
...looking at this closer, this actually doesn't encode the way we need. This encodes an object with the string key inside it, but the format we want actually needs part of the data to be in the key itself:

```
"extendedData": [
    "2_0": {
        "colspan": 2,
        "rowspan": 0
    },
    "2_1": {
        "colspan": 0,
        "rowspan": 1
    },
    "1_1": {
        "colspan": 0,
        "rowspan": 1
    },
    "1_0": {
        "colspan": 2,
        "rowspan": 2
    }
],
```

Is it possible to have the `Codable` implementation reach into the parent's keys like that?

Reply:
Ah. I see the difference in the encoded data now. The `Set` encodes each element as an object. I don't know if there's a way to flatten that structure without wrapping the set (to do custom encoding) or doing custom encoding at the `Table` level ...

Reply:
That encodes the extended data as a without individual JSON objects for each element:
```
"extendedData" : {
  "2_1" : {
    "colspan" : 0,
    "rowspan" : 1
  },
  "1_0" : {
    "colspan" : 2,
    "rowspan" : 2
  },
  "2_0" : {
    "colspan" : 2,
    "rowspan" : 0
  },
  "1_1" : {
    "colspan" : 0,
    "rowspan" : 1
  }
}
```

Reply:
I like that suggestion, i've pushed up a commit that includes it. I also went ahead and rebased onto the latest `main`.